### PR TITLE
Adding realtime segment metrics

### PR DESCRIPTION
**Pull request by fx19880617**
Adding realtime segment metrics for number of segments and number of documents.

Current NotifyCommit method call will do closeSegment then addSegment, which will do segment number -1 then +1 which means the realtime segment metrics are not updated any way.

### CODE REVIEW

**Comment by kishoreg** [comment]
LGTM. Subbu, I am merging this. I am assuming this should not interfere with incrementing/decrementing read/write locks

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java [hunk 1]
```diff
@@ -303,7 +314,7 @@ public void run() {
     });
 
     indexingThread.start();
-
+    serverMetrics.addValueToTableGauge(tableName, ServerGauge.SEGMENT_COUNT, 1L);
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java]: If we stop indexing (due to a shutdown), then we need to decrement the count by one, right?

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java [hunk 2]
```diff
@@ -176,6 +185,7 @@ public void run() {
             throw e;
           }
         } while (notFull && keepIndexing && (!isShuttingDown));
+
         if (isShuttingDown) {
           segmentLogger.info("Shutting down indexing thread!");
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java]: decrement segment count here?

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java [hunk 3]
```diff
@@ -341,6 +352,13 @@ private void computeKeepIndexing() {
         keepIndexing = false;
       }
     }
+    updateCurrentDocumentCountMetrics();
```
Comment (mcvsubbu) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java]: Not here, I would do it after we finish creating the index segment (but before we call notifySegmentCommitted)

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java (hunk 1 above)
Comment (fx19880617) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java]: There are two scenarios we will hit realtime segment shutting down:
1. server shutdown
2. realtime table deletion.
I think in both cases, we should be fine.

File: pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java (hunk 3 above)
Comment (fx19880617) [inline pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java]: This is to periodically update the metrics inside segmentStatusTask, which will be running every 1mins. 
So we will have number of documents metric updated in realtime segment.

**Comment by fx19880617** [comment]
close as already pushed.

**Pull request closed without merging**